e2e/cypress/integration/archived_channel/archive_channel_search_spec.js

```diff
@@ -0,0 +1,163 @@
+        cy.get('#channelHeaderTitle').should('be.visible');
+
+        // # Select Archive Channel from the header menu
+        cy.uiArchiveChannel();
+
+        // # Archive dialogue message reads"This will archive the channel from the team and make its contents inaccessible for all users" (Mobile dialogue makes no mention of the data will be accessible)
+        cy.get('#searchBox').focus().clear();
+        cy.get('#searchBox').focus().type(`${testArchivedMessage}{enter}`);
+
+        // * Post is not returned by search
+        cy.get('#searchContainer').should('be.visible');
+        cy.get('.no-results__wrapper').should('be.visible');
+    });
+
+    it('MM-T1709 Archive a channel while search results are displayed in RHS', () => {
+        const messageText = `search ${getRandomId()} pineaples`;
+
+        // # Post a unique string of text in a channel
+        cy.uiCreateChannel({prefix: 'archive-while-searching'}).then(() => {
+            cy.postMessage(messageText);
+
+            // # Search for the string of text from step 1
+            cy.get('#searchBox').focus().clear();
+            cy.get('#searchBox').focus().type(`${messageText}{enter}`);
+
+            // * Post is returned by search, since it's not archived anymore
+            cy.get('#searchContainer').should('be.visible');
+
+            // # Observe the post is displayed in RHS results
+            cy.get('.search-item-snippet').first().should('contain.text', messageText);
+
+            // # While the RHS is still open with the results, archive the channel the post was made in
+            cy.uiArchiveChannel();
+            cy.get('#searchContainer').should('be.visible');
+            cy.get('.no-results__wrapper').should('be.visible');
+        });
+    });
+    it('MM-T1710 archived channels are not listed on the "in:" autocomplete', () => {
+        // # Archive a channel and make a mental note of the channel name
+
+        // # Place the focus in the search box
+        cy.get('#searchBox').focus().clear();
+
+        // # Type "in:" and note the list of channels that appear
+        cy.get('#searchBox').focus().type(`in:${testChannel.name}`);
+        cy.findByTestId(testChannel.name).should('not.exist');
+    });
+});
+
+function createArchivedChannel(channelOptions, messages, memberUsernames) {
+    let channelName;
+    return cy.uiCreateChannel(channelOptions).then((newChannel) => {
+        channelName = newChannel.name;
+        if (memberUsernames) {
+            cy.uiAddUsersToCurrentChannel(memberUsernames);
+        }
+        if (messages) {
+            let messageList = messages;
+            if (!Array.isArray(messages)) {
+                messageList = [messages];
+            }
+            messageList.forEach((message) => {
+                cy.postMessage(message);
+            });
+        }
+        cy.uiArchiveChannel();
+
+        return cy.wrap({name: channelName});
+    });
+}
```

Comment:
I've seen this being used in multiple spec files. I suggest to move this in a common file, by convention as `helpers.js`.